miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultWorkflowExecutor.java

```diff
@@ -30,4 +57,52 @@ public Pool save(Pool pool) throws IOException {
   public SequencerPartitionContainer save(SequencerPartitionContainer spc) throws IOException {
     return containerService.save(spc);
   }
+
+  @Override
+  public QC save(QC qc) throws IOException {
+    return qualityControlService.save(qc);
+  }
+
+  @Override
+  public Collection<QcType> getQcTypeList() throws IOException {
+    return qualityControlService.listQcTypes();
+  }
+
+  @Override
+  public Sample save(Sample sample) throws IOException {
+    return sampleService.save(sample);
+  }
+
+  @Override
+  public SampleAliquot createAliquotFromParent(Sample stock) throws IOException {
+    SampleAliquot aliquot = new SampleAliquotImpl();
+    DetailedSample sample = (DetailedSample) stock;
+
+    aliquot.setScientificName(sample.getScientificName());
+    aliquot.setSampleType(sample.getSampleType());
+    aliquot.setParent(sample);
+    if (sample.getSubproject() != null) aliquot.setSubproject(sample.getSubproject());
+    aliquot.setGroupId(sample.getGroupId());
+    aliquot.setGroupDescription(sample.getGroupDescription());
+    aliquot.setSampleClass(sampleClassService.getAll().stream()
+        .filter(sampleClass -> {
+          try {
+            return sampleClass.getSampleCategory().equals("Aliquot") && sampleValidRelationshipService.getAll().stream()
+                .anyMatch(
+                    validRelationship -> !validRelationship.getArchived() && validRelationship.getChild().getId() == sampleClass.getId()
+                    && validRelationship.getParent().getId() == sample.getSampleClass().getId());
+          } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)